### PR TITLE
generate load dags with none or some specific tag(s)

### PR DESCRIPTION
**Pull request by fupelaqu**
## Summary
generate load dags with none or some specific tag(s)

**PR Type: Feature**

**Status: Ready to review**

**Breaking change? No**

### CODE REVIEW

File: src/main/scala/ai/starlake/schema/generator/DagGenerateCmd.scala [hunk 1]
```diff
@@ -68,14 +68,10 @@ object DagGenerateCmd extends Cmd[DagGenerateConfig] {
   ): Try[JobResult] =
     Try {
       val cmd = new DagGenerateCommand(schemaHandler)
-      if (config.domains) {
```
Comment (hayssams) [inline src/main/scala/ai/starlake/schema/generator/DagGenerateCmd.scala]: What's the added value of this change ?

File: src/main/scala/ai/starlake/schema/generator/DagGenerateCmd.scala [hunk 2]
```diff
@@ -68,14 +68,10 @@ object DagGenerateCmd extends Cmd[DagGenerateConfig] {
...
+      if (config.domains || !config.tasks) {
         cmd.generateDomainDags(config)
       }
-      if (config.tasks) {
-        cmd.generateTaskDags(config)
-      }
-      if (!config.tasks && !config.domains) {
-        cmd.generateDomainDags(config)
+      if (config.tasks || !config.domains) {
```
Comment (hayssams) [inline src/main/scala/ai/starlake/schema/generator/DagGenerateCmd.scala]: And this one ?

File: src/main/scala/ai/starlake/schema/generator/DagGenerateCommand.scala [hunk 3]
```diff
@@ -26,11 +26,12 @@ class DagGenerateCommand(schemaHandler: SchemaHandler) extends LazyLogging {
...
 
   private def tableWithDagConfigs(
-    dagConfigs: Map[String, DagGenerationConfig]
+    dagConfigs: Map[String, DagGenerationConfig],
+    tags: Set[String]
   )(implicit settings: Settings): List[TableWithDagConfig] = {
     logger.info("Starting to generate dags")
     val tableWithDagConfigAndSchedule = schemaHandler.domains().flatMap { domain =>
-      domain.tables.flatMap { table =>
+      domain.tables.filter(tags.isEmpty || _.tags.intersect(tags).nonEmpty).flatMap { table =>
```
Comment (hayssams) [inline src/main/scala/ai/starlake/schema/generator/DagGenerateCommand.scala:34]: 👌

**Review by hayssams — COMMENTED** [review]
Gr8 to have tags working on domains too

**Review by hayssams — APPROVED** [review]

File: src/main/scala/ai/starlake/schema/generator/DagGenerateCmd.scala (hunk 2 above)
Comment (fupelaqu) [reply src/main/scala/ai/starlake/schema/generator/DagGenerateCmd.scala]: None, I will revert those changes

**Pull request merged**